nixos/modules/services/misc/zoneminder.nix

```diff
@@ -63,9 +63,14 @@ let
     ${cfg.extraConfig}
   '';
 
-  phpExtensions = with pkgs.phpPackages; [
-    { pkg = apcu; name = "apcu"; }
-  ];
+  phpPackage =
+    let
+      base = pkgs.php;
+    in
+      base.buildEnv {
+        extensions = e: with e;
+          base.enabledExtensions ++ [ apcu ];
+      };
```

Comment:
Since you're already using `buildEnv` instead of `withExtensions` you're off to a good start when it comes to the phpOptions as you have defined below.

Look here: https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/php.section.md#installing-php-with-packages

And you can see how you can customize the options as well.

Other than that I would think that it should work, unless you're based on a too old master because we had a bug in the original PR that has been fixed in a later version of master. That fix comes from this PR: https://github.com/NixOS/nixpkgs/pull/84721

Reply:
Is using
`services.phpfpm.pools.<name>.phpPackage = php.buildEnv { extraConfig="..." }`
preferred over using
`services.phpfpm.pools.<name>.phpOptions = "..."`
or does it not matter?

Reply:
Both should work, but the buildEnv already builds a config and use it, so it's one config file less which is nice.

Reply:
Both should be fine - go for what you think is most readable :) If you go for `services.phpfpm.pools.<name>.phpOptions`, you should probably use `php.withExtensions` instead of `php.buildEnv`, though.

Reply:
I personally think `withExtensions` + `phpOptions` is most readable. Just pushed this change. Thanks @etu and @talyz